src/main/java/Bob/Parser.java

```diff
@@ -0,0 +1,125 @@
+package bob;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Locale;
+
+import bob.task.Deadline;
+import bob.task.Event;
+import bob.task.Task;
+import bob.task.Todo;
+
+class Parser {
+
+    /**
+     * Parses a string into a LocalDateTime object.
+     *
+     * @param input
+     * @return LocalDateTime
+     * @throws BobException
+     */
+    public static LocalDateTime parseDateTime(String input) throws BobException {
+        try {
+            // Correct format: "dd/MM/yyyy HHmm" (e.g., "02/12/2019 1800")
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm", Locale.ENGLISH);
+            return LocalDateTime.parse(input, formatter);
+        } catch (DateTimeParseException e) {
+            throw new BobException("The date and time format is incorrect: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Command to list all tasks.
+     *
+     * @param tasks
+     */
+    public static void processListCommand(List<Task> tasks) {
+        String out = "Here are the tasks in your list: \n";
+        for (int i = 0; i < tasks.size(); i++) {
+            out += (i + 1) + ". " + tasks.get(i) + "\n";
+        }
+        Ui.dialogue(out);
+    }
+
+    /**
+     * Command to mark, unmark, or delete a task.
+     *
+     * @param userInput
+     * @param tasks
+     * @throws BobException
+     * @throws IOException
+     */
+    public static void processTaskModificationCommands(String userInput, List<Task> tasks) throws BobException, IOException {
```

Comment:
processTaskModificationCommand, while explained in the header, is rather confusing and unintuitive... Perhaps you might want to consider creating separate static methods or even classes for them?